app/views/users/discussions.html.erb

```diff
@@ -14,17 +14,25 @@
         <%= t :discussions_will_be_here %>
       </div>
     <% else %>
-      <table class="table table-striped">
-        <% @watched_discussions.each do |discussion| %>
-          <tr>
-            <td>
-              <%= icon_for_read(discussion.read_by?(@user)) %>
-            </td>
-            <td><%= link_to discussion.item.name, item_discussion_path(discussion) %></td>
-            <td><%= time_ago_in_words discussion.last_message_date %></td>
-          </tr>
-        <% end %>
-      </table>
+      <div class="table-responsive mb-3">
+        <table class="table">
+          <% @watched_discussions.each do |discussion| %>
+            <% if discussion.read_by?(@user) != @last_read %>
+              <%= user_discussions_table_title(discussion, @user, @last_read) %>
+              <% @last_read = discussion.read_by?(@user) %>
```

Comment:
I'm aware this looks a little bit funny, but this is so `unread` and `read` messages are separated as in the screenshot. Also works when there is only one type of message on the page (for example only `read`), as in the other screenshot.

Other ideas included grouping all `@watched_discussions` into two arrays, one for unread and one for read, and adding titles for each as long as those arrays were not empty. But this seems simpler than that and with almost everything extracted to helpers it ends up ok, I think.